components/dashboard/sections/overview/Welcome.tsx

```diff
@@ -239,16 +252,18 @@ const WelcomeCategoryButton = ({ img, title, description, className, onClick, st
       )}
       onClick={onClick}
     >
-      <div className="absolute top-2 right-2 flex cursor-pointer items-center rounded-full bg-white px-2 py-1 text-xs text-slate-600">
-        {isCompleted ? (
-          <Check size={12} />
-        ) : (
-          <React.Fragment>
-            <ListCheck size={12} className="mr-1" /> {completedSteps}/{steps.length}
-          </React.Fragment>
-        )}
-      </div>
-      {img}
+      {steps && (
+        <div className="absolute top-2 right-2 flex cursor-pointer items-center rounded-full bg-white px-2 py-1 text-xs text-slate-600">
+          {isCompleted ? (
+            <Check size={12} />
+          ) : (
+            <React.Fragment>
+              <ListCheck size={12} className="mr-1" /> {completedSteps}/{steps?.length}
```

Comment:
Looks like you added a null check above, should be safe already

```suggestion
              <ListCheck size={12} className="mr-1" /> {completedSteps}/{steps.length}
```